### PR TITLE
Enhancement/3773 move idea hub widget footer into widget.footer prop

### PR DESCRIPTION
**Pull request by danielgent**
## Summary

Addresses issue #3773

https://user-images.githubusercontent.com/8496063/128244019-700ba937-d422-41bb-994f-e5f947180f73.mp4

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-ideas-widget.scss [hunk 1]
```diff
@@ -263,4 +258,12 @@
...
 		}
 	}
+
+	.googlesitekit-widget--ideaHubIdeas .googlesitekit-widget__footer {
+		padding: 14px 24px;
+	}
+
+	.googlesitekit-widget--ideaHubIdeas .googlesitekit-idea-hub__footer {
+		padding: 0;
+	}
```
Comment (danielgent) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-ideas-widget.scss:268]: This reset not mentioned in IB.  Padding is being applied to this by a `.mdc-layout-grid` class

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js [hunk 2]
```diff
@@ -65,25 +101,19 @@ const Pagination = ( {
...
 	);
 };
 
 Pagination.propTypes = {
-	total: PropTypes.number.isRequired,
-	page: PropTypes.number.isRequired,
-	ideasPerPage: PropTypes.number,
-	handlePrev: PropTypes.func,
-	handleNext: PropTypes.func,
+	tab: PropTypes.string,
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:112]:
```suggestion
	tab: PropTypes.oneOf( [ 'new-ideas', 'saved-ideas', 'draft-ideas' ] ),
```

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js [hunk 3]
```diff
@@ -26,20 +26,52 @@ import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
...
+		if ( page > 1 ) {
+			setValue( uniqueKey, page - 1 );
+		}
+	}, [ page, setValue, uniqueKey ] );
+
+	const handleNext = useCallback( () => {
+		if ( page < Math.ceil( total / IDEA_HUB_IDEAS_PER_PAGE ) ) {
+			setValue( uniqueKey, page + 1 );
+		}
+	}, [ page, setValue, total, uniqueKey ] );
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:73]: Let's not display the pagination if there are no ideas:

```suggestion
	}, [ page, setValue, total, uniqueKey ] );

	if ( total < 1 ) {
		return null;
	}
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @danielgent. Looks good to me. Added three minor suggestions that I'll apply by myself.

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js [hunk 4]
```diff
@@ -109,7 +115,7 @@ const Pagination = ( { tab } ) => {
 };
 
 Pagination.propTypes = {
-	tab: PropTypes.string,
+	tab: PropTypes.oneOf( [ 'new-ideas', 'saved-ideas', 'draft-ideas' ] ),
```
Comment (danielgent) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:118]: I'm surprised these aren't constants defined somewhere :smile: 

We could also define `PropTypes.oneOf( [ 'new-ideas', 'saved-ideas', 'draft-ideas' ] )` in a constants file somewhere.  `react/prop-types` gets confused by this but we're not using that anywhere

**Pull request merged**